docs/kratos/emails-sms/01_sending-emails-smtp.mdx

```diff
@@ -241,6 +263,35 @@ The courier passes the following object as the `ctx` parameter into the Jsonnet
 
 In most cases, the default payload should be sufficient.
 
+### HTTP Integrations
+
+Sample configurations to send emails via HTTP using different providers.
+
+#### Sendgrid
+
+Use the following connection URI and settings to
+[send emails using Sendgrid](https://www.twilio.com/docs/sendgrid/api-reference/mail-send/mail-send) via HTTP.
+
+```
+...
+courier:
+  delivery_strategy: http
+  http:
+    request_config:
+      url: https://api.sendgrid.com/v3/mail/send
+      method: POST
+      body: file:///etc/config/kratos/mail.template.jsonnet
```

Comment:
What's in this file?

Reply:
For example something like 

```
function(ctx) {
"personalizations": [
    {
      "to": [
        {
          "email": if "TemplateData" in ctx && "To" in ctx.TemplateData then ctx.TemplateData.To else null)
        }
      ],
     "verificationCode": if "TemplateData" in ctx && "VerificationCode" in ctx.TemplateData then ctx.TemplateData.VerificationCode else null
    }
  ],
  // Other values and personalizations .......
}
```

It might only be relevant for self-hosting? Not super familiar with that part unfortunately.

Reply:
Nah, example needs to be here!

Reply:
Added. 

I don't quite get what this is though - wouldnt you use HTML templates to customize the emails normally?  
Or is this something else? Haven't seen a mail JSONNET template anywhere else, and seems we dont have any examples for

Reply:
The json net sets the HTTP body of the request. Since the sendgrid API expects JSON we need to transform the context (template data etc) to JSON first.